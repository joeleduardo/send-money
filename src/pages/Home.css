@font-face {
  font-family: 'Alverata';
  src: url('../assets/Alverata-Black.otf') format('woff')
}

body {
  font-family: Helvetica, Arial, sans-serif;
}

h1, h3 {
  font-family: 'Alverata', Helvetica, Arial, sans-serif;
}

section {
  padding: 2rem;
  border-bottom: 1px solid #cccccc;
}

section:last-of-type {
  border-bottom: none;
}

.snackbar {
  position: absolute;
  width: 100%;
  margin-top: -1rem;
}

.snackbar div {
  border-radius: 8px;
  color: #ffffff;
  padding: 1rem;
  background-color: #333333;
}

.no-items {
  padding: 1rem;
  color: #999999;
  background-color: #efefef;
}

.wrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  margin: 2rem auto;
}

@media screen and (min-width: 850px) {
  section {
    width: 100%;
    border-right: 1px solid #cccccc;
    border-bottom: none;
  }

  section:last-of-type {
    border-right: none;
  }

  .wrapper {
    flex-direction: row;
    max-width: 980px;
  }
}