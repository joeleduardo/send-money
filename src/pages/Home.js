import React from 'react';
import { firestore } from '../firebase'
import PropTypes  from 'prop-types';
import Input from '../components/Input';
import Button from '../components/Button';
import List from '../components/List';
import Chart from '../components/Chart';
import './Home.css';

class Home extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      amountAvailable: 5000,
      amountSent: 0,
      amountValidationMsg: '',
      formValid: false,
      validationErrors: {
        name: '',
        email: '',
        amount: ''
      }
    }
  }

  componentDidMount() {
    this.props.getLoans(firestore.collection('loans').get());
  }

  componentDidUpdate(prevProps) {
    if (this.props.data.loans !== prevProps.data.loans) {
      const totalLoans = this.props.data.loans.reduce((total, amount) => {
        total += parseFloat(amount.amount)
        return total
      }, 0);

      this.setState({
        amountAvailable: this.state.amountAvailable - totalLoans,
        amountSent: totalLoans
      }, () => this.cleanFields());
    }
  }

  cleanFields () {
    this.setState({
      name: '',
      email: '',
      amount: '',
      formValid: false,
      validationErrors: {
        name: '',
        email: '',
        amount: ''
      }
    })
  }

  validateField(field, value) {
    const validationErrors = this.state.validationErrors;
    let name;
    let email;
    let amount;

    switch (field) {
      case 'name':
        name =  value.trim().length === 0 || value.length < 3 ? 'Please enter a valid name' : '';
        validationErrors.name = name;
        break
      case 'email':
        email = !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ? 'Please enter a valid email address' : '';
        validationErrors.email = email;
        break
      case 'amount':
        amount = value.trim().length === 0 || parseFloat(value) === 0 || isNaN(Number(value)) ? 'Please enter a valid amount' : '';
        validationErrors.amount = amount;
        break
      default:
        break
    }

    this.setState({
      validationErrors
    }, () => {
      const fields = !!(this.state.name && this.state.email && this.state.amount)
      const validations = (!this.state.validationErrors.name && !this.state.validationErrors.email && !this.state.validationErrors.amount)
      this.setState({
        formValid: (fields && validations)
      })
    })
  }

  inputHandler (e) {
    const {
      value,
      name
    } = e.target;

    this.setState({
      [name]: value
    }, () => this.validateField(name, value));
  }

  timer() {
    setTimeout(() => {
      this.setState({
        amountValidationMsg: ''
      })
    }, 2000)
  }

  submitForm () {
    const {
      name,
      email,
      amount,
    } = this.state;

    const amountInt = parseFloat(amount).toFixed(2);
    const amountValidation = amountInt <= this.state.amountAvailable

    if (amountValidation) {
      const postTransaction = firestore.collection('loans').add({name, email, amount: amountInt});
      const getTransaction = firestore.collection('loans').get();
      this.props.addLoan(postTransaction, getTransaction);
    } else {
      window.scroll({ top: 0, left: 0, behavior: 'smooth' });
      this.setState({
        amountValidationMsg: 'You don\'t have enough money to send it'
      }, () => this.timer());
    }
  }

  render() {
    const chartLabels = ['Total sent', 'Left available']
    const chartData = [this.state.amountSent.toFixed(2), this.state.amountAvailable.toFixed(2)]

    return(
      <div className='wrapper'>
        {this.state.amountValidationMsg &&
        <div className='snackbar'>
          <div>{this.state.amountValidationMsg}</div>
        </div>
        }
        <section>
          <h1>Send Money</h1>
          <div>
            <Input name='name' validation={this.state.validationErrors.name} label='Name' value={this.state.name} prompt='Friend name' onChange={e => this.inputHandler(e)} />
            <Input name='email' validation={this.state.validationErrors.email} label='Email' value={this.state.email} prompt='friend@friend.com' onChange={e => this.inputHandler(e)} />
            <Input name='amount' validation={this.state.validationErrors.amount} label='Amount' value={this.state.amount} prompt='1000.00' currency='Â£' onChange={e => this.inputHandler(e)} />
            <Button text='Send' onClick={e => this.submitForm(e)} enabled={this.state.formValid}/>
          </div>
        </section>
        <section>
          <h1>My Account</h1>
          <Chart chartLabels={chartLabels} chartData={chartData}/>
          <div>
            <h3>Transactions</h3>
            {this.props.data.loans.length > 0 ? (
              <List items={this.props.data.loans}/>
            ):(
              <div className='no-items'>No transaction registered!</div>
            )}
          </div>
        </section>
      </div>
    )
  }
}

Home.propTypes = {
  data: PropTypes.object,
  getLoans: PropTypes.func,
  addLoan:PropTypes.func
}

Home.defaultProps = {
  data: {},
  getLoans: () => {},
  addLoan: () => {}
}

export default Home;
