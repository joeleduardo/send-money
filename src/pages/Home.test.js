import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

const data = {
  loans: []
}

jest.useFakeTimers();

test('Home page', () => {
  const cHome = shallow(<Home data={data}/>);

  cHome.setState({
    amountAvailable: 5000
  })

  cHome.setProps({
    data: {
      loans: [
        {
          name: 'Joel Ortiz',
          email: 'joeleduardo@hotmail.com',
          amount: '1000.00'
        }
      ]
    }
  });

  const nameEmptyEvent = {
    target: {
      name: 'name',
      value: ''
    }
  };

  const emailEmptyEvent = {
    target: {
      name: 'email',
      value: ''
    }
  };

  const amountEmptyEvent = {
    target: {
      name: 'amount',
      value: ''
    }
  };

  cHome.find('Input').at(0).simulate('change', nameEmptyEvent);
  cHome.find('Input').at(1).simulate('change', emailEmptyEvent);
  cHome.find('Input').at(2).simulate('change', amountEmptyEvent);

  const nameEvent = {
    target: {
      name: 'name',
      value: 'Joel Ortiz'
    }
  };

  const emailEvent = {
    target: {
      name: 'email',
      value: 'joeleduardo@hotmail.com'
    }
  };

  const amountEvent = {
    target: {
      name: 'amount',
      value: '100'
    }
  };

  cHome.find('Input').at(0).simulate('change', nameEvent);
  cHome.find('Input').at(1).simulate('change', emailEvent);
  cHome.find('Input').at(2).simulate('change', amountEvent);
  cHome.instance().validateField('', '');

  expect(cHome.state().validationErrors).toEqual({
    name: '',
    email: '',
    amount: ''
  });

  cHome.setState({
    name: 'Joel Ortiz',
    email: 'joeleduardo@hotmail.com',
    amount: '100.00'
  })

  cHome.find('Button').simulate('click');
  expect(parseFloat(cHome.state().amount)).toBeLessThanOrEqual(cHome.state().amountAvailable)

  cHome.setState({
    name: 'Samantha Adriano',
    email: 'sammadrianos@gmail.com',
    amount: '5500.00'
  })

  cHome.find('Button').simulate('click');
  expect(parseFloat(cHome.state().amount)).toBeGreaterThan(cHome.state().amountAvailable);

  jest.runAllTimers()
  expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 2000);
  expect(cHome).toMatchSnapshot();
});

