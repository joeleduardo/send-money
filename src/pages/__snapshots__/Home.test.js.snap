// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    addLoan={[Function]}
    data={
        Object {
            "loans": Array [
              Object {
                "amount": "1000.00",
                "email": "joeleduardo@hotmail.com",
                "name": "Joel Ortiz",
              },
            ],
          }
    }
    getLoans={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <section>
          <h1>
                    Send Money
          </h1>
          <div>
                    <Input
                              currency=""
                              label="Name"
                              name="name"
                              onChange={[Function]}
                              prompt="Friend name"
                              validation=""
                              value="Samantha Adriano"
                    />
                    <Input
                              currency=""
                              label="Email"
                              name="email"
                              onChange={[Function]}
                              prompt="friend@friend.com"
                              validation=""
                              value="sammadrianos@gmail.com"
                    />
                    <Input
                              currency="£"
                              label="Amount"
                              name="amount"
                              onChange={[Function]}
                              prompt="1000.00"
                              validation=""
                              value="5500.00"
                    />
                    <Button
                              enabled={true}
                              onClick={[Function]}
                              text="Send"
                    />
          </div>
</section>,
        <section>
          <h1>
                    My Account
          </h1>
          <chartComponent
                    chartData={
                              Array [
                                        "1000.00",
                                        "4000.00",
                                      ]
                    }
                    chartLabels={
                              Array [
                                        "Total sent",
                                        "Left available",
                                      ]
                    }
          />
          <div>
                    <h3>
                              Transactions
                    </h3>
                    <List
                              items={
                                        Array [
                                                  Object {
                                                    "amount": "1000.00",
                                                    "email": "joeleduardo@hotmail.com",
                                                    "name": "Joel Ortiz",
                                                  },
                                                ]
                              }
                    />
          </div>
</section>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Send Money
</h1>,
            <div>
              <Input
                            currency=""
                            label="Name"
                            name="name"
                            onChange={[Function]}
                            prompt="Friend name"
                            validation=""
                            value="Samantha Adriano"
              />
              <Input
                            currency=""
                            label="Email"
                            name="email"
                            onChange={[Function]}
                            prompt="friend@friend.com"
                            validation=""
                            value="sammadrianos@gmail.com"
              />
              <Input
                            currency="£"
                            label="Amount"
                            name="amount"
                            onChange={[Function]}
                            prompt="1000.00"
                            validation=""
                            value="5500.00"
              />
              <Button
                            enabled={true}
                            onClick={[Function]}
                            text="Send"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send Money",
            },
            "ref": null,
            "rendered": "Send Money",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Input
                  currency=""
                  label="Name"
                  name="name"
                  onChange={[Function]}
                  prompt="Friend name"
                  validation=""
                  value="Samantha Adriano"
/>,
                <Input
                  currency=""
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  prompt="friend@friend.com"
                  validation=""
                  value="sammadrianos@gmail.com"
/>,
                <Input
                  currency="£"
                  label="Amount"
                  name="amount"
                  onChange={[Function]}
                  prompt="1000.00"
                  validation=""
                  value="5500.00"
/>,
                <Button
                  enabled={true}
                  onClick={[Function]}
                  text="Send"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "",
                  "label": "Name",
                  "name": "name",
                  "onChange": [Function],
                  "prompt": "Friend name",
                  "validation": "",
                  "value": "Samantha Adriano",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "",
                  "label": "Email",
                  "name": "email",
                  "onChange": [Function],
                  "prompt": "friend@friend.com",
                  "validation": "",
                  "value": "sammadrianos@gmail.com",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "£",
                  "label": "Amount",
                  "name": "amount",
                  "onChange": [Function],
                  "prompt": "1000.00",
                  "validation": "",
                  "value": "5500.00",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "enabled": true,
                  "onClick": [Function],
                  "text": "Send",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              My Account
</h1>,
            <chartComponent
              chartData={
                            Array [
                                          "1000.00",
                                          "4000.00",
                                        ]
              }
              chartLabels={
                            Array [
                                          "Total sent",
                                          "Left available",
                                        ]
              }
/>,
            <div>
              <h3>
                            Transactions
              </h3>
              <List
                            items={
                                          Array [
                                                        Object {
                                                          "amount": "1000.00",
                                                          "email": "joeleduardo@hotmail.com",
                                                          "name": "Joel Ortiz",
                                                        },
                                                      ]
                            }
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Account",
            },
            "ref": null,
            "rendered": "My Account",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "chartData": Array [
                "1000.00",
                "4000.00",
              ],
              "chartLabels": Array [
                "Total sent",
                "Left available",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Transactions
</h3>,
                <List
                  items={
                                    Array [
                                                      Object {
                                                        "amount": "1000.00",
                                                        "email": "joeleduardo@hotmail.com",
                                                        "name": "Joel Ortiz",
                                                      },
                                                    ]
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Transactions",
                },
                "ref": null,
                "rendered": "Transactions",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "items": Array [
                    Object {
                      "amount": "1000.00",
                      "email": "joeleduardo@hotmail.com",
                      "name": "Joel Ortiz",
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <section>
            <h1>
                        Send Money
            </h1>
            <div>
                        <Input
                                    currency=""
                                    label="Name"
                                    name="name"
                                    onChange={[Function]}
                                    prompt="Friend name"
                                    validation=""
                                    value="Samantha Adriano"
                        />
                        <Input
                                    currency=""
                                    label="Email"
                                    name="email"
                                    onChange={[Function]}
                                    prompt="friend@friend.com"
                                    validation=""
                                    value="sammadrianos@gmail.com"
                        />
                        <Input
                                    currency="£"
                                    label="Amount"
                                    name="amount"
                                    onChange={[Function]}
                                    prompt="1000.00"
                                    validation=""
                                    value="5500.00"
                        />
                        <Button
                                    enabled={true}
                                    onClick={[Function]}
                                    text="Send"
                        />
            </div>
</section>,
          <section>
            <h1>
                        My Account
            </h1>
            <chartComponent
                        chartData={
                                    Array [
                                                "1000.00",
                                                "4000.00",
                                              ]
                        }
                        chartLabels={
                                    Array [
                                                "Total sent",
                                                "Left available",
                                              ]
                        }
            />
            <div>
                        <h3>
                                    Transactions
                        </h3>
                        <List
                                    items={
                                                Array [
                                                            Object {
                                                              "amount": "1000.00",
                                                              "email": "joeleduardo@hotmail.com",
                                                              "name": "Joel Ortiz",
                                                            },
                                                          ]
                                    }
                        />
            </div>
</section>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Send Money
</h1>,
              <div>
                <Input
                                currency=""
                                label="Name"
                                name="name"
                                onChange={[Function]}
                                prompt="Friend name"
                                validation=""
                                value="Samantha Adriano"
                />
                <Input
                                currency=""
                                label="Email"
                                name="email"
                                onChange={[Function]}
                                prompt="friend@friend.com"
                                validation=""
                                value="sammadrianos@gmail.com"
                />
                <Input
                                currency="£"
                                label="Amount"
                                name="amount"
                                onChange={[Function]}
                                prompt="1000.00"
                                validation=""
                                value="5500.00"
                />
                <Button
                                enabled={true}
                                onClick={[Function]}
                                text="Send"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send Money",
              },
              "ref": null,
              "rendered": "Send Money",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Input
                    currency=""
                    label="Name"
                    name="name"
                    onChange={[Function]}
                    prompt="Friend name"
                    validation=""
                    value="Samantha Adriano"
/>,
                  <Input
                    currency=""
                    label="Email"
                    name="email"
                    onChange={[Function]}
                    prompt="friend@friend.com"
                    validation=""
                    value="sammadrianos@gmail.com"
/>,
                  <Input
                    currency="£"
                    label="Amount"
                    name="amount"
                    onChange={[Function]}
                    prompt="1000.00"
                    validation=""
                    value="5500.00"
/>,
                  <Button
                    enabled={true}
                    onClick={[Function]}
                    text="Send"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currency": "",
                    "label": "Name",
                    "name": "name",
                    "onChange": [Function],
                    "prompt": "Friend name",
                    "validation": "",
                    "value": "Samantha Adriano",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currency": "",
                    "label": "Email",
                    "name": "email",
                    "onChange": [Function],
                    "prompt": "friend@friend.com",
                    "validation": "",
                    "value": "sammadrianos@gmail.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "currency": "£",
                    "label": "Amount",
                    "name": "amount",
                    "onChange": [Function],
                    "prompt": "1000.00",
                    "validation": "",
                    "value": "5500.00",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "enabled": true,
                    "onClick": [Function],
                    "text": "Send",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                My Account
</h1>,
              <chartComponent
                chartData={
                                Array [
                                                "1000.00",
                                                "4000.00",
                                              ]
                }
                chartLabels={
                                Array [
                                                "Total sent",
                                                "Left available",
                                              ]
                }
/>,
              <div>
                <h3>
                                Transactions
                </h3>
                <List
                                items={
                                                Array [
                                                                Object {
                                                                  "amount": "1000.00",
                                                                  "email": "joeleduardo@hotmail.com",
                                                                  "name": "Joel Ortiz",
                                                                },
                                                              ]
                                }
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Account",
              },
              "ref": null,
              "rendered": "My Account",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "chartData": Array [
                  "1000.00",
                  "4000.00",
                ],
                "chartLabels": Array [
                  "Total sent",
                  "Left available",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Transactions
</h3>,
                  <List
                    items={
                                        Array [
                                                            Object {
                                                              "amount": "1000.00",
                                                              "email": "joeleduardo@hotmail.com",
                                                              "name": "Joel Ortiz",
                                                            },
                                                          ]
                    }
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Transactions",
                  },
                  "ref": null,
                  "rendered": "Transactions",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "items": Array [
                      Object {
                        "amount": "1000.00",
                        "email": "joeleduardo@hotmail.com",
                        "name": "Joel Ortiz",
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
